csharp/src/Ice/IObject.cs

```diff
@@ -6,26 +6,24 @@
 using System;
 using System.ComponentModel;
 using System.Diagnostics;
+using System.Threading;
 using System.Threading.Tasks;
 
 namespace Ice
 {
-    /// <summary>
-    /// Interface for incoming requests.
-    /// </summary>
-    public interface IRequest
-    {
-        /// <summary>
-        /// Returns the {@link Current} object for this the request.
-        /// </summary>
-        /// <returns>The Current object for this request.</returns>
-        Current? GetCurrent();
-    }
-
     ///<summary>The base interface for all servants.</summary>
     public interface IObject
     {
-        public Task<OutputStream>? Dispatch(Incoming inS, Current current)
+        /// <summary>Dispatches a request on this servant.</summary>
+        /// <param name="istr">The <see cref="InputStream"/> that holds the request frame. It is positionned at
+        /// the start of the request's payload, just after starting to read the encapsulation.</param>
+        /// <param name="current">The current parameter holds decoded header data and other information about the
+        /// current request.</param>
+        /// <returns>A value task that provides the response frame for the request. Null means the request is a oneway
+        /// request. See <see cref="IceInternal.Protocol.StartResponseFrame"/>.</returns>
+        /// <exception cref="System.Exception">Any exception thrown by Dispatch is marshaled into the response frame.
+        /// </exception>
+        public ValueTask<OutputStream>? Dispatch(InputStream istr, Current current)
```

Comment:
Did you consider to return a non nullable task with an empty frame for one-ways?

Reply:
I didn't consider it. For oneway requests, there is no response, so such an empty response frame would be discarded. Say we always return the same empty never-returned response frame in this situation. I think it would give the impression we're sending back a response frame for all requests, when we're not.

Note the AMD mapping and generated IceD_ methods provide similar return types:
- (IceD) ValueTask<OutputStream> for non-void operations
- (IceD) ValueTask<OutputStream>? for void operations (null == oneway)
- (AMD) Task? for AMD methods (null == oneway or synchronous success with void return value)

Reply:
my only concern is that you cannot do `await Dispatcher` that is the standard way to call an API returning a Task, I never come across a .NET API that return null tasks

Reply:
That's a good point, I'll change it.

Reply:
Fixed. Dispatch is now DispatchAsync:
```
public ValueTask<OutputStream> DispatchAsync(InputStream istr, Current current)
```
The AMD methods return:
- ValueTask<TResult> for non-void return
- ValueTask for void return

All the generated IceD_ methods have an Async suffix and return a ValueTask<OutputStream>. Furthermore, all IceD_ methods that adapt an AMD (XxxAsync) method are now marked async and await the AMD method.